### PR TITLE
PART C- Documentation and Comments

### PR DESCRIPTION
**Pull request by gagandeep8030**
Hi Everyone,

Part C is completed.
Below mentioned are the files modified by the respective team member.

README.md was modified by both of the team members and the work was divided equally.

Gagandeep - include/data_structures/message.hpp
                       test/src/receiver/main.cpp

Shubham - include/atomics/receiverCadmium.hpp
                    test/src/sender/main.cpp

Kindly review this pull request and suggest changes if necessary.
Thanks

### CODE REVIEW

**Comment by Namithajacob** [comment]
In receivercadmium.hpp line 91, you have to add curly braces.

**Comment by shubhamthakur463** [comment]
@Namithajacob 
Changes suggested by you have been added. Kindly review this pull request again.
Thanks.

**Comment by dilsgeorge** [comment]
all the input and output paths are hard coded in main.cpp of top_model,receiver,sender ("../test/data/receiver/receiver_test_output.txt"). This needs to be soft coded ,use #define

File: test/src/receiver/main.cpp [hunk 1]
```diff
@@ -122,7 +162,11 @@ int main(){
...
   	);
 
-	/*********************************************/
+	/**
+     * In the following code model and runner are created with their 
+     * corresponding elasped time.
+     * After which simulation starts and is ran until 04:00:00:000 time.
+     */
 
   	auto elapsed1 = std::chrono::duration_cast<std::chrono::duration<double,
```
Comment (dilsgeorge) [inline test/src/receiver/main.cpp]: kindly give meaningful name for elapsed1 and elapsed

**Comment by Namithajacob** [comment]
You need to check the main.cpp of receiver and sender and receivercadmium for spelling mistakes.

**Comment by gagandeep8030** [comment]
@Namithajacob Could you please more specific about the line number for the correction required regarding your last comment. For your future reference, kindly tag the specific line of the code where you want necessary changes for better understanding.

**Comment by Namithajacob** [comment]
> @Namithajacob Could you please more specific about the line number for the correction required regarding your last comment. For your future reference, kindly tag the specific line of the code where you want necessary changes for better understanding.



> @Namithajacob Could you please more specific about the line number for the correction required regarding your last comment. For your future reference, kindly tag the specific line of the code where you want necessary changes for better understanding.

@gagandeep8030 Since there were many changes to be done, i thought it would be better if you could go through the entire documentation once. But in future i will definitely tag the specific line

File: README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+**Developed By**: Cristina Ruiz Martin<br/>
+**Modified By**:Gagandeep Singh and Shubham Thakur
+---
+### PROJECT DESCRIPTION
+
+ABP (Alternating Bit Protocol) is a communication protocol to ensure reliable transmission through unreliable network. The sender sends a packet and waits for an acknowledgement. If the acknowledgement doesn't arrive within a predefined time, the sender re-sends this packet until it receives an expected acknowledgement and then sends the next packet. In order to distinguish two consecutive packets, the sender adds an additional bit on each packet (called alternating bit because the sender uses 0 and 1 alternatively).
+
+The ABP Simulator consists of 3 components: sender, network and receiver.
+
+1. **Receiver** - It receives the data and sends back acknowledgement after certain perios of time.
```
Comment (Namithajacob) [inline README.md]: kindly check the spelling for period

File: README.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+2. **Subnets** -  It passes the packets after a time delay. In order to simulate the unreliability of the network, only 95% of the packets will be passed in each of the subnet, i.e. 5% of the data will be lost through the subnet.
+3. **Sender** - Its behavior is pretty complex. The sender changes from initial phase *passive* to *active* when an external signal is received. Once activated it start sending the packet with the alternating bit. Every time a packet is sent, it waits for an acknowledgement during an specific waiting time. It the acknowledgment does not arrive within the time window, the sender will re- send the previous packet with the alternating bit. If the expected acknowledgement is received within the time window, the sender will send the next packet. When there are no more packets to send, the sender will go again to the *passive* phase.
```
Comment (Namithajacob) [inline README.md]: consider changing during an to for

Comment (shubhamthakur463) [reply README.md]: Changes have been made. Kindly check again.

File: README.md (hunk 2 above)
Comment (shubhamthakur463) [reply README.md]: Suggested changes have been made.

**Comment by dilsgeorge** [comment]
Since there is README.md , i think we don't need README.txt also

File: include/atomics/receiver_cadmium.hpp [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    /**
+	 * Function responsible for the internal transition.
+     * Sending state of the receiver is turned off here.
+	 */
+    void internal_transition() {
+        state.sending = false; 
+    }
+
+    /**
+	 * Function to perform the external transition.
```
Comment (dilsgeorge) [inline include/atomics/receiver_cadmium.hpp]: i think its better to give @params and @return for function comments in doxygen format for better description

Comment (gagandeep8030) [reply include/atomics/receiver_cadmium.hpp]: Required changed have been made. Kindly review the changes.

**Comment by shubhamthakur463** [comment]
Hi reviewers, 
It has been 5 days since we created this pull request. All the suggested changes have been made. Kindly review and approve it as soon as possible. There are only two days for the project submission deadline.
Thanks.

**Pull request merged**